VideoLocker/src/main/java/org/edx/mobile/view/custom/ECheckedTextView.java

```diff
@@ -0,0 +1,40 @@
+package org.edx.mobile.view.custom;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.Typeface;
+import android.util.AttributeSet;
+import android.widget.CheckedTextView;
+
+import org.edx.mobile.R;
+
+public class ECheckedTextView extends CheckedTextView {
+    public ECheckedTextView(Context context) {
+        super(context);
+    }
+
+    public ECheckedTextView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        processAttrs(context, attrs);
+    }
+
+    public ECheckedTextView(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        processAttrs(context, attrs);
+    }
+
+    private void processAttrs(Context context, AttributeSet attrs) {
+        if (isInEditMode()) return;
+        TypedArray a = context.getTheme().obtainStyledAttributes(
+                attrs, R.styleable.custom_view, 0, 0);
+        // Check for the font attribute and setup font
+        String fontFileName = a.getString(R.styleable.custom_view_font);
+        a.recycle();
+        if (fontFileName == null) {
+            // If font is not defined, then set default font
+            fontFileName = "OpenSans-Regular.ttf";
```

Comment:
here's a hack-ish way of applying font app-wide using **reflection**:
http://stackoverflow.com/a/16883281/1402616
I've used it in an app, and it gets the job done with minimal amount of lines written and code changes.
What I like in this solution is that there isn't a need to set font in each style or widget declaration.
_The only downside I see is, we can use only upto 4 fonts within an app._